packages/compass-components/src/components/editor.tsx

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+
+if (typeof window === 'undefined' && typeof globalThis !== 'undefined') {
+  // ace-builds wants to install itself on `window`, which
+  // is not available when this package is loaded through
+  // (non-Electron) Node.js. That's an atypical case, but it's
+  // easier to account for it here than to handle all cases
+  // in which this package is loaded from Node.js.
```

Comment:
What are the cases where we load this in Node.js environment with no window available? In mocha test envs it should be provided by jsdom, in electron and browser it should exist 🤔

Reply:
@gribnoysup Was running into this with gen-esm-wrapper in databases-collections and was not a fan of trying to remove it there for this, but let me know if you prefer that :sweat_smile:

Reply:
Gotcha! I think it's fine to keep then, wasn't obvious where it can happen and so I thought that maybe this was actually highlighting a real issue (like running tests in wrong env or something)